include/etl/iterator.h

```diff
@@ -604,7 +604,191 @@ namespace etl
+
+  /**
+   * @brief  Turns assignment into insertion.
+   *
+   * These are output iterators, constructed from a container-of-T.
+   * Assigning a T to the iterator appends it to the container using push_back.
+   *
+   * @tparam TContainer
+   */
+  template <class TContainer>
+  class back_insert_iterator : public iterator<output_iterator_tag, void, void, void, void>
+  {
+  public:
+    /// A nested typedef for the type of whatever container you used.
+    typedef TContainer container_type;
+
+    /// The only way to create this %iterator is with a container.
+    explicit ETL_CONSTEXPR17 back_insert_iterator(TContainer& c)
+      : container(addressof(c))
+    {
+    }
+
+    /**
+     * This kind of %iterator doesn't really have a @a position in the
+     * container (you can think of the position as being permanently at
+     * the end, if you like).  Assigning a value to the %iterator will
+     * always append the value to the end of the container.
+     *
+     * @param value An instance of whatever type container_type::const_reference is;
+     *              presumably a reference-to-const T for container<T>.
+     * @return This %iterator, for chained operations.
+     */
+    ETL_CONSTEXPR17 back_insert_iterator& operator=(const typename TContainer::value_type& value)
+    {
+      container->push_back(value);
+      return (*this);
+    }
+
+#if ETL_USING_CPP11
+    ETL_CONSTEXPR17 back_insert_iterator& operator=(typename TContainer::reference value)
+    {
+      container->push_back(move(value));
+      return (*this);
+    }
+#endif  // ETL_USING_CPP11
+
+    /// Simply returns *this.
+    ETL_NODISCARD ETL_CONSTEXPR17 back_insert_iterator& operator*()
+    {
+      return (*this);
+    }
+
+    /// Simply returns *this.  (This %iterator does not @a move.)
+    ETL_CONSTEXPR17 back_insert_iterator& operator++()
+    {
+      return (*this);
+    }
+
+    /// Simply returns *this.  (This %iterator does not @a move.)
+    ETL_CONSTEXPR17 back_insert_iterator operator++(int)
+    {
+      return (*this);
+    }
+
+  protected:
+    TContainer* container;
+  };
+
+  /**
+   * This wrapper function helps in creating back_insert_iterator instances.
+   * Typing the name of the %iterator requires knowing the precise full
+   * type of the container, which can be tedious and impedes generic
+   * programming.  Using this function lets you take advantage of automatic
+   * template parameter deduction, making the compiler match the correct types for you.
+   *
+   * @tparam TContainer The container type.
+   * @param container A container of arbitrary type.
+   * @return An instance of back_insert_iterator working on @p container.
+   */
+  template <class TContainer>
+  ETL_NODISCARD ETL_CONSTEXPR17 inline ETL_OR_STD::back_insert_iterator<TContainer> back_inserter(TContainer& container)
+  {
+    return ETL_OR_STD::back_insert_iterator<TContainer>(container);
```

Comment:
`ETL_OR_STD` is used for a small set of functions and classes that may be expected when interacting with the the STL, especially when (Argument Dependent Lookup (ADL) is invoked in the ETL.
e.g.
std::swap()
std::begin()
std::end()

In the above case in think you should just be returning the ETL version.

Reply:
Let me resume the usage of `ETL_OR_STD` :

Definition of `ETL_OR_STD` depends on `ETL_NOT_USING_STL` itself dependings on `ETL_NO_STL`.

According to the [documentation](https://www.etlcpp.com/no_stl.html) :
> When `ETL_NO_STL` is defined then the ETL will use its own versions, otherwise the STL versions will be used.

In this case, I think that the actual code is the right think to do. **Is that correct ?**